lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/PIDStoreHandler.cs

```
@@ -0,0 +1,79 @@
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+
+using Org.Apache.REEF.Common.Files;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.Common.Runtime.Evaluator
+{
+    /// <summary>
+    /// A PID store handler for c# evaluator
+    /// </summary>
+    public sealed class PIDStoreHandler
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(PIDStoreHandler));
+
+        private bool _pidIsWritten = false;
+
+        private object lockObject = new object();
+
+        private REEFFileNames _reefFileNames;
+
+        [Inject]
+        private PIDStoreHandler(REEFFileNames reefFileNames)
+        {
+            _reefFileNames = reefFileNames;
+        }
+
+        /// <summary>
+        /// Writes PID to the runtime folder
+        /// </summary>
+        public void WritePID()
+        {
+            lock (lockObject)
+            {
+                if (!_pidIsWritten)
+                {
+                    string currentDirectory = Directory.GetCurrentDirectory();
+                    string path = Path.Combine(currentDirectory, _reefFileNames.GetPidFileName());
+
+                    try
+                    {
+                        var pid = Process.GetCurrentProcess().Id;
+                        using (StreamWriter sw = File.CreateText(path))
+                        {
+                            sw.WriteLine(pid);
+                            sw.Flush();
+                        }
+
+                        Logger.Log(Level.Verbose, "Writing PID {0} to file {1}", pid, path);
+                        _pidIsWritten = true;
+                    }
+                    catch (Exception e)
+                    {
+                        Logger.Log(Level.Error, "Failed writing PID with exception {0}", e);                        
```

Is it OK to ignore this exception? Under which circumstances can this fail in a way that isn't a bug in our code?